starks/floatingpoint.py

```diff
@@ -0,0 +1,189 @@
+import random
+from starks.euclidean import gcd
+from starks.euclidean import extended_euclidean_algorithm
+from starks.polynomial import polynomials_over
+from starks.poly_utils import generate_irreducible_polynomial
+from starks.poly_utils import generate_primitive_polynomial
+from starks.modp import IntegersModP
+from starks.numbertype import FieldElement
+from starks.numbertype import DomainElement
+from starks.numbertype import memoize
+from starks.numbertype import typecheck
+from starks.poly_utils import is_irreducible
+from starks.finitefield import FiniteField
+
+#this function generates a binary list of 2^index
+def num_to_binary_list_pow(index):
+  output_list = []
+
+  for i in range(index):
+    output_list.append(0)
+  output_list.append(1)
+
+  return output_list 
+
+
+#this function generates a binary list represents index
+def num_to_binary_list(index):
+  output_list = []
+
+  while index > 0:
+    output_list.append(index%2)
+    index = index // 2  
+
+  return output_list
+
+
+#less than and equality operations between numbers in binary finite field representation
+def LQ(self, other):
```

Comment:
I think `self` shouldn't be here since this function isn't inside a class. Should this be shifted into the class below?

Reply:
Perhaps just rename to be `x`, `y` instead of `self`, `other` for clarity.

Reply:
I addressed this comment.